### PR TITLE
chmod -R a+w /srv

### PR DESCRIPTION
**Pull request by danielballan**

### CODE REVIEW

File: binder/postBuild [hunk 1]
```diff
@@ -2,3 +2,6 @@
 
 pip install git+https://github.com/bluesky/tiled.git@v0.1.0a15#egg=tiled[complete]
 pip install git+https://github.com/bluesky/databroker.git@da8c8a65297137b51dc8206a614287a72950767a#egg=databroker[complete]
+
+# This is here so that the software environment can be changed in a tmpfs overlay.
+chmod -R a+w /srv
```
Comment (mrakitin) [inline binder/postBuild:7]: Maybe making it more verbose for potential future debugging?
```suggestion
chmod -Rv a+w /srv
```

**Review by mrakitin — APPROVED** [review]
Looks good to me, just a minor suggestion.

**Comment by danielballan** [comment]
Unfortunately this doesn't work because `binder/postBuild` runs as `$NB_USER`, not `root`. Exploring alternatives...

**Comment by mrakitin** [comment]
`sudo chmod ...`? It worked for the CI for `profile_collection`s on Azure Pipelines.

**Comment by danielballan** [comment]
No, `$NB_USER` does not have `sudo`.

File: binder/appendix.txt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+USER root
+# Enable user to edit software environment via a writeable tmpfs overlay
+# in Singularity.
+RUN chmod a+w -R /srv
```
Comment (mrakitin) [inline binder/appendix.txt:4]: Maybe `chown`-ing to `${NB_USER}` will eliminate the need for a world-writable directory? Not sure if there are any limitations though.

Comment (danielballan) [reply binder/appendix.txt:4]: The `${NB_USER}`, known at build time, is `jovyan`, but the container is run as whatever BNL `$USER` is logged in, which will have a different uid.

**Comment by danielballan** [comment]
Summary:

* I created an "appendix" that switches to `root`, runs the `chmod ...` command, and then switches back to `NB_USER`.
* Since `jupyter-repo2docker` supports `--appendix` but the official `repo2docker-action` GH Action does not expose that, I extended the `repo2docker-action` on a fork under the NSLS-II org to expose it. I opened a [PR to submit the enhancement upstream](https://github.com/jupyterhub/repo2docker-action/pull/72) so that, if accepted, we can stop running on a custom fork.

Evidence it worked from [the build logs](https://github.com/NSLS-II/scipy-binder/runs/2764234989?check_suite_focus=true#step:4:4610):

```
Show Variables
  DOCKER_REGISTRY:
  INPUT_ADDITIONAL_TAG: 
  INPUT_APPENDIX_FILE: binder/appendix.txt
  INPUT_BINDER_CACHE: 
  INPUT_BINDER_CACHE: 
  INPUT_IMAGE_NAME: danielballan/scipy-binder
  INPUT_IMAGE_NAME: danielballan/scipy-binder
  INPUT_MYBINDERORG_TAG: 
  INPUT_MYBINDERORG_TAG: 
  INPUT_NOTEBOOK_USER: 
  INPUT_NO_PUSH: true
  INPUT_PUBLIC_REGISTRY_CHECK: 
  INPUT_REPO_DIR: 
  NB_USER: jovyan
  PWD: /github/workspace
  SHA_NAME: danielballan/scipy-binder:b474ec74b909
```

```
Validate Information
  Appendix read from binder/appendix.txt:
  USER root
  # Enable user to edit software environment via a writeable tmpfs overlay
  # in Singularity.
  RUN chmod a+w -R /srv
  USER ${NB_USER}
  Using default tag: latest
```

```
  Step 53/56 : CMD ["jupyter", "notebook", "--ip", "0.0.0.0"]
   ---> Running in b57fd8913f99
  Removing intermediate container b57fd8913f99
   ---> 7bde39bb4bfe
  Step 54/56 : USER root
   ---> Running in 5107b0af7f3e
  Removing intermediate container 5107b0af7f3e
   ---> 6f96ef092a90
  Step 55/56 : RUN chmod a+w -R /srv
   ---> Running in c5a1e992ddd0
  Removing intermediate container c5a1e992ddd0
   ---> 9fd5cc90605a
  Step 56/56 : USER ${NB_USER}
   ---> Running in 1def2330a694
  Removing intermediate container 1def2330a694
   ---> 5bbed7afa3f3
  {"aux": {"ID": "sha256:5bbed7afa3f3312660b1c00c8942c0cbf2c65858d789680bf692b403e216186b"}}Successfully built 5bbed7afa3f3
  Successfully tagged danielballan/scipy-binder:b474ec74b909
```

File: .github/workflows/build_image.yml [hunk 3]
```diff
@@ -9,6 +9,7 @@ jobs:
       uses: actions/checkout@main
 
     - name: update jupyter dependencies with repo2docker
-      uses: jupyterhub/repo2docker-action@6a93d1bf173f32cfd859130d5851aa8a7aad07cd  # master as of 2020-10-20
+      uses: NSLS-II/repo2docker-action@c64bda966539ff36856055311be7d709605a8a97  # adds support for --appendix
```
Comment (mrakitin) [inline .github/workflows/build_image.yml:12]: Nice addition! https://github.com/jupyterhub/repo2docker-action/pull/72 is for future reference.

**Review by mrakitin — APPROVED** [review]
Looks good to me!

**Pull request merged**